src/ripple/app/misc/HashRouter.cpp

```diff
@@ -49,12 +49,19 @@ HashRouter::addSuppression(uint256 const& key)
 
 bool
 HashRouter::addSuppressionPeer(uint256 const& key, PeerShortID peer)
+{
+    auto [added, _] = addSuppressionPeerWithStatus(key, peer);
+    return added;
+}
+
+std::pair<bool, std::optional<Stopwatch::time_point>>
+HashRouter::addSuppressionPeerWithStatus(const uint256& key, PeerShortID peer)
```

Comment:
The `HashRouter` API was already confusing enough. It's now become a little _too_ confusing. I don't have an improvement to offer, but we need to think about this.